### PR TITLE
Add `async_stub` method

### PR DESCRIPTION
**Pull request by PerchunPak**
Which actually the same as `stub` but creates async stub.

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -110,3 +110,7 @@ It may receive an optional name that is shown in its ``repr``, useful for debugg
 
         foo(stub)
         stub.assert_called_once_with('foo', 'bar')
+
+.. seealso::
+
+    ``async_stub`` method, which actually the same as ``stub`` but makes async stub.
```
Comment (PerchunPak) [inline docs/usage.rst:116]: image

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Thanks a lot @PerchunPak for the contribution, sorry about the delay. 

There are some things that need fixing:

1. Can you take a look at the test failures? Seems like we have to put a guard around the definition of `async_stub` on the Python versions that don't support it.
2. Could you add a very simple test to `TestMockerStub` that just calls `async_stub` and ensures it is an instance of `unittest.mock.AsyncMock`? This is just to avoid future regressions. The test needs to be skipped on unsupported Python versions.
3. Finally could add a quick changelog entry for this, under a new session `3.8.0 (unreleased)`?

Thanks again!

**Comment by PerchunPak** [comment]
> Can you take a look at the test failures? Seems like we have to put a guard around the definition of async_stub on the Python versions that don't support it.

Can't imagine how, seems like `mock_module.AsyncMock` will not work with type checkers, so how this could be done?
Maybe import standalone `mock` module in `if TYPE_CHECKING:` block?

**Comment by nicoddemus** [comment]
Sorry I should have mention, but [mypy supports `sys.version_info` checks](https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks). 👍

**Comment by PerchunPak** [comment]
> Sorry I should have mentioned, but [mypy supports `sys.version_info` checks](https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks). 👍

Yes, but how about runtime? Yes `unittest.mock` doesn't have `AsyncMock` in py3.7, but `mock` have.

**Comment by nicoddemus** [comment]
I'm OK with only supporting AsyncMock for Python 3.8+, or is this something you need in Python 3.7?

**Comment by PerchunPak** [comment]
> I'm OK with only supporting AsyncMock for Python 3.8+, or is this something you need in Python 3.7?

No, just you support both (`unittest.mock` and `mock`), it's strange that you breaking this support here.

**Comment by nicoddemus** [comment]
Hmm an alternative would be then:

```python
if sys.version_info[:2] >= (3, 8):
    AsyncMockType = unittest.mock.AsyncMock
else:
    AsyncMockType = mock.AsyncMock
```

at the top of the file, then use `AsyncMockType` in the annotation. If the above does not work, then `AsyncMockType = Any` for Python 3.7 is a good compromise.

**Comment by nicoddemus** [comment]
Awesome, thanks a lot for the contribution and patience! 🙇

**Pull request merged**